pkg/plugins/discovery.go

```diff
@@ -37,25 +38,25 @@ func (l *LocalRegistry) Plugin(name string) (*Plugin, error) {
 	filepath := filepath.Join(l.path, name)
 	specpath := filepath + ".spec"
 	if fi, err := os.Stat(specpath); err == nil {
```

Comment:
It feels weird that `readPluginSocketInfo(socketpath, fi)` receives both a path string and a File info. You should probably refactor this to be `readPluginSocketInfo(socketpath) (*Plugin, err)` and then do a:

```go
if socket, err = readPluginSocketInfo(socketpath); err == nil {
return socket
}```